src/ui/public/kuery/lib/functions/not.js

```diff
@@ -0,0 +1,38 @@
+import * as ast from '../ast';
+import { nodeTypes } from '../node_types';
+
+export function buildNodeParams(child, serializeStyle = 'operator') {
+  return {
+    arguments: [child],
+    serializeStyle
+  };
+}
+
+export function toElasticsearchQuery(node, indexPattern) {
+  let [ argument ] = node.arguments;
+  if (argument.type === 'literal') {
+    argument = nodeTypes.function.buildNode('is', '*', argument.value);
+  }
+
+  return {
+    bool: {
+      must_not: ast.toElasticsearchQuery(argument, indexPattern)
+    }
+  };
+}
+
+export function toKueryExpression(node) {
+  if (node.serializeStyle !== 'operator') {
```

Comment:
Yes, that's exactly the argument I made about `:` above :)   I wonder if you're looking at an outdated version of the PR? I think at one point I categorized `:` as `shorthand`, but I've since updated it to `operator`.

I'll try to sum up my thoughts here since I can't do an inline reply to your review summary below. I don't believe I've used these terms inconsistently. I think we're just operating with different definitions in mind. Here's how I categorize things:

* `implicit` (just `and`)
* `operator` (`foo:bar`, `foo or bar`, `foo and bar`, `-foo`) (basically anything the grammar has a special case for)
* `function` (`is(foo, bar)`)
* `shorthand` (just range `bytes:[1000 to 8000]`)

I wouldn't be surprised if these categorizations evolved along with the language, but they seem to work ok for now.